runtime/parachains/src/scheduler.rs

```diff
@@ -391,7 +362,7 @@ impl<T: Config> Pallet<T> {
 	pub(crate) fn next_up_on_time_out(core: CoreIndex) -> Option<ScheduledCore> {
 		Self::next_up_on_available(core).or_else(|| {
 			// Or, if none, the claim currently occupying the core,
-			// as it would be put back on the queue after timing out.
+			// as it would be put back on the queue after timing out if number of retries is not at the maximum.
```

Comment:
I might be missing something, but it does not seem we check that. We just assume it is true and always return the currenctly occupying para.

Reply:
I'm not following. I just extended the comment in code with what you commented on here.

Reply:
I was not talking about the comment. The way I understand the code, it does not do what the comment says.

Reply:
[Here](https://github.com/paritytech/polkadot/blob/7309626063046bd0c0423de330a4ae05032f24e4/runtime/parachains/src/scheduler.rs#L369) we always return the currently occupying para, no matter the current retry count, while in fact we should return `None` it is already at the maximum. That would at least be my understanding.